test/metabase/query_processor/middleware/catch_exceptions_test.clj

```diff
@@ -104,37 +104,50 @@
                  :metadata
                  (update :stacktrace boolean)))))))
 
-(deftest include-query-execution-info-test
-  (testing "Should include info from QueryExecution if added to the thrown/raised Exception"
+(deftest catch-exceptions-test
+  (testing "include-query-execution-info-test"
+    (testing "Should include info from QueryExecution if added to the thrown/raised Exception"
+      (tu.log/suppress-output
+       (is (= {:status     :failed
+               :class      java.lang.Exception
+               :error      "Something went wrong"
+               :stacktrace true
+               :card_id    300
+               :json_query {}
+               :row_count  0
+               :data       {:cols []}
+               :a          100
+               :b          200}
+              (-> (mt/test-qp-middleware catch-exceptions/catch-exceptions
+                                         {} {} []
+                                         {:runf (fn [_ _ context]
+                                                  (qp.context/raisef (ex-info "Something went wrong."
+                                                                              {:query-execution {:a            100
+                                                                                                 :b            200
+                                                                                                 :card_id      300
+                                                                                                 ;; these keys should all get removed
+                                                                                                 :result_rows  400
+                                                                                                 :hash         500
+                                                                                                 :executor_id  500
+                                                                                                 :dashboard_id 700
+                                                                                                 :pulse_id     800
+                                                                                                 :native       900}}
+                                                                              (Exception. "Something went wrong"))
+                                                                     context))})
+                  :metadata
+                  (update :stacktrace boolean)))))))
+  (testing "Should always include :error (#23258, #23281)"
     (tu.log/suppress-output
```

Comment:
`tu.log/suppress-output` is deprecated but running the tests in the repl still would print the exceptions so it still has a purpose. It was deprecated now that we have a log config which sets the log threshold to fatal but still has a use for ad-hoc repl testing, ie `(catch-exceptions-test)` at the repl.

Reply:
can't we just... put in a temp config in test context, since this PR is not in the repl?

Reply:
We do have a test config: https://github.com/metabase/metabase/blob/master/test_config/log4j2-test.xml . I'm addressing when you actually run these tests in the repl while developing. In this context the normal log4js.xml is used so the errors get logged and they end up in the repl.